simplejson/encoder.py

```diff
@@ -39,6 +39,22 @@ def _import_speedups():
 
 FLOAT_REPR = repr
 
+class _JsonStr(str):
```

Comment:
It would be better not to make this a str/unicode subclass in the first place, if it's just wrapping some str or unicode object then there's no copying involved, just the allocation of this "tag". It also should greatly simplify the code. Something as simple as this would suffice:

```python
class RawJSON(object):
    def __init__(self, encoded_json):
        self.encoded_json = encoded_json
```